staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1426,35 +1434,120 @@ func (c *cacheWatcher) process(ctx context.Context, resourceVersion uint64) {
 	}
 }
 
+// ready is a condition variable that represent readiness and
```

Comment:
this grew quite a bit - can you maybe move that to a separate file (like pkg/storage/cache/ready.go and ready_test.go)

Reply:
ack